contracts/PrizeTierHistory.sol

```diff
@@ -1,182 +1,97 @@
 // SPDX-License-Identifier: GPL-3.0
 pragma solidity 0.8.6;
-
 import "@pooltogether/owner-manager-contracts/contracts/Manageable.sol";
 import "./interfaces/IPrizeTierHistory.sol";
+import "./abstract/BinarySearchLib.sol";
 
-/**
- * @title  PoolTogether V4 IPrizeTierHistory
- * @author PoolTogether Inc Team
- * @notice IPrizeTierHistory is the base contract for PrizeTierHistory
- */
 contract PrizeTierHistory is IPrizeTierHistory, Manageable {
-    /* ============ Global Variables ============ */
-    /**
-     * @notice History of PrizeTier updates
-     */
-    PrizeTier[] internal history;
-
-    /* ============ Constructor ============ */
-    constructor(address _owner) Ownable(_owner) {}
 
-    /* ============ External Functions ============ */
+    using BinarySearchLib for uint32[];
 
-    // @inheritdoc IPrizeTierHistory
-    function push(PrizeTier calldata _nextPrizeTier) external override onlyManagerOrOwner {
-        PrizeTier[] memory _history = history;
+    uint32[] internal history;
+    mapping(uint32 => PrizeTier) internal prizeTiers;
```

Comment:
Add documentation explaining both of these fields.  The history needs to explain how it's an array of draw ids, and the prizeTiers is mapping needs to explain whether it's mapping by history index or drawId.